### PR TITLE
DEV: move `discourse_dev` gem to the core.

### PR DESCRIPTION
**Pull request by vinothkannans**
It uses the newly created gem [`discourse_dev_assets`](https://github.com/discourse/discourse_dev_assets).

It has the rake helper tasks for Discourse developers. These tasks can be used to populate random data like topics, groups, users, categories, and tags in your local discourse_development database.

### CODE REVIEW

File: lib/tasks/dev.rake [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  require 'highline/import'
+
+  answer = ask("Do you want to repopulate the database with fresh data? It will recreate DBs and run migration from scratch before generating all the samples. (Y/n)  ")
+
+  if (answer == "" || answer.downcase == 'y')
+    Rake::Task['dev:reset'].invoke
+  else
+    puts "You can run `bin/rails dev:reset` to repopulate anytime."
+  end
+end
```
Comment (davidtaylorhq) [inline lib/tasks/dev.rake:47]: Do we need `check_environment!` calls for these tasks to prevent them being run in production?

**Review by davidtaylorhq — APPROVED** [review]
Looks great to me - just the one comment about the rake tasks

File: lib/tasks/dev.rake (hunk 1 above)
Comment (vinothkannans) [reply lib/tasks/dev.rake:47]: I think so. I don't want admins to execute the rake task accidentally which will trigger `db:migrate:reset` without any check,

Comment (davidtaylorhq) [reply lib/tasks/dev.rake:47]: Makes sense! So I think we need to add `check_environment!` to `dev:config`, `dev:populate` and `dev:repopulate`? (it is missing from those commands at the moment)

Comment (vinothkannans) [reply lib/tasks/dev.rake:47]: No, those tasks will fail anyway without the `faker` gem. And they won't trigger `db:migrate:reset` task. So it's safe.

**Pull request merged**